pci: Fix deletion of PCI address identifier config

cathay4t opened the pull request:
When deleting config of PCI address identifier interface via YAML:

```
---
interfaces:
  - name: wan0
    type: ethernet
    state: absent
    identifier: pci-address
    pci-address: 0000:07:00.0
```

Nmstate failed to delete so because nmstate discard the `pci-address`
during deserialization. Fixed by include `pci-address` for `absent`
interface during deserialization.

Integration test case included.

ihuguet commented on rust/src/lib/iface.rs:
```diff
@@ -315,6 +315,9 @@ impl<'de> Deserialize<'de> for Interface {
             if let Some(s) = v.get("mac-address") {
                 new_value.insert("mac-address".to_string(), s.clone());
             }
+            if let Some(s) = v.get("pci-address") {
+                new_value.insert("pci-address".to_string(), s.clone());
+            }
```
Some lines above these ones there is a comment `// Ignore all properties except type if state: absent`. It's clearly outdated, please change it.
Apart from this, LGTM.

cathay4t replied:
Amended that comment.

cathay4t committed:
pci: Fix deletion of PCI address identifier config

When deleting config of PCI address identifier interface via YAML:

```
---
interfaces:
  - name: wan0
    type: ethernet
    state: absent
    identifier: pci-address
    pci-address: 0000:07:00.0
```

Nmstate failed to delete so because nmstate discard the `pci-address`
during deserialization. Fixed by include `pci-address` for `absent`
interface during deserialization.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
